callouts/java/service-callout/src/main/java/service/ServiceCallout.java

```diff
@@ -0,0 +1,234 @@
+package service;
+
+/*
+ * Copyright 2015 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.envoyproxy.envoy.service.ext_proc.v3.*;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+import io.grpc.netty.NettyServerBuilder;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static utils.SslUtils.createSslContext;
+import static utils.SslUtils.readFileToBytes;
+
+/**
+ * Server that manages startup/shutdown of a {@code Greeter} server.
+ */
+public abstract class ServiceCallout {
+    private static final Logger logger = Logger.getLogger(ServiceCallout.class.getName());
+
+    private Server server;
+
+    public ServiceCallout() {
+        this(null, null, null, null, null, null, null, null, null, null, null, null, null);
+    }
+
+    public ServiceCallout(
```

Comment:
This is a good baseline example, I am having some thoughts on how this could be made into a more useful decorator style interface. Something to allow both the default constructor to be called and for settings to be modified in an overloaded constructor. 

What do you think?
I can put together an example when I have some free cycles.